liquibase-standard/src/main/java/liquibase/changelog/ChangeSet.java

```diff
@@ -776,6 +776,22 @@ public ExecType execute(DatabaseChangeLog databaseChangeLog, ChangeExecListener
         return execType;
     }
 
+    //
+    // If this is for MSSQL and we have an AbstractSqlChange
+    // then we add an appending SQL visitor that will add
+    // on the end delimiter if necessary
+    //
+    private List<SqlVisitor> addSqlVisitorsIfNecessary(Database database, Change change) {
+        if (! (database instanceof MSSQLDatabase && change instanceof AbstractSQLChange) || ((AbstractSQLChange)change).getEndDelimiter() != null) {
```

Comment:
This isn't just an AbstractSQLChange thing, is it? We're wanting all changes on mssql to have an end delimter? Removing that part of the `if` makes it more globally applicable and so less surprising to users, and also keeps us from adding more dependencies from database/change generic code like ChangeSet to implementation details.

Also, maybe better to change this from a "if not this case, just return getSqlVisitors()" and instead have it as "if this case, add to getSqlVisitors()"? Maybe easier to read now, and easier to maintain later if/when we want to add other visitors for other databases?

Finally, in theory the ChangeSet shouldn't have dependencies on MSSQL either, since that's more of an implementation. Maybe instead of adding this MSSQL-specific if block here, the MSSQLDatabase.executeStatements() should have the logic of adding the delimiter?